apps/src/dance/danceRedux.ts

```diff
@@ -172,6 +174,9 @@ const danceSlice = createSlice({
     ) => {
       state.currentAiModalField = action.payload;
     },
+    setAiModalOpenedFromFlyout: (state, action: PayloadAction<boolean>) => {
```

Comment:
Should we combine this with `setCurrentAiModalField` into a single `openAiModal(state, action: PayloadAction<{modalField: Field, fromFlyout: boolean}>)` so you always have to specify where the modal was opened from when opening? That way we minimize the risk of forgetting to set `aiModalOpenedFromFlyout` since it's not a separate action. Can also be accompanied by a `closeAiModal(state)` which automatically sets `currentAiModalField` to `undefined` and `aiModalOpenedFromFlyout` to `false`?

Reply:
Updated here, LMK what you think! https://github.com/code-dot-org/code-dot-org/pull/54629/commits/bde8ace2a770f21c17306cdd0bce36496b2b3a6e